### PR TITLE
Vp9 enc on cnl

### PR DESCRIPTION
**Pull request by xhaihao**
This patch series adds support for vp9 encoding on CNL

### CODE REVIEW

File: src/gen10_vdenc_common.c [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+#include <string.h>
+#include <math.h>
+#include <assert.h>
+
+#include "intel_batchbuffer.h"
+#include "i965_defines.h"
+#include "i965_drv_video.h"
+#include "gen10_vdenc_common.h"
+
+#define VDENC_WRITE_COMMANDS(command_flag)                        \
```
Comment (QuPengfei) [inline src/gen10_vdenc_common.c]: it is good add *param and batch as the Macro define parameter
define VDENC_WRITE_COMMANDS(command_flag,param,batch)

File: src/gen10_vdenc_vp9.h [hunk 2]
```diff
@@ -0,0 +1,1644 @@
...
+#define VDENC_VDENC_CMD0_STATE_SIZE_IN_DWS              (VDENC_VDENC_CMD0_STATE_SIZE >> 2)
+#define VDENC_VDENC_CMD1_STATE_SIZE_IN_DWS              (VDENC_VDENC_CMD1_STATE_SIZE >> 2)
+
+enum VDENC_VP9_LUT_MODE {
+    VDENC_VP9_LUT_MODE_INTRA_SADMPM             = 0x00,
+    VDENC_VP9_LUT_MODE_INTRA_32x32              = 0x01,
+    VDENC_VP9_LUT_MODE_INTRA_16x16              = 0x02,
+    VDENC_VP9_LUT_MODE_INTRA_8x8                = 0x03,
+
+    VDENC_VP9_LUT_MODE_INTER_32x16              = 0x04,
```
Comment (QuPengfei) [inline src/gen10_vdenc_vp9.h]: these three emum value use the same 0x04. Is it reasonable?

Comment (xhaihao) [reply src/gen10_vdenc_vp9.h]: This enum isn't used any more, I will remove it from this commit.

File: src/gen10_vdenc_common.c (hunk 1 above)
Comment (xhaihao) [reply src/gen10_vdenc_common.c]: Thanks for the comment, I will update the commit to redefine this Macro.

**Comment by xhaihao** [comment]
@QuPengfei thanks for your review, I Updated the corresponding commits per your comments.

File: src/i965_defines.h [hunk 3]
```diff
@@ -911,6 +911,7 @@
 #define HEVC_SLICE_I                            2
 
 #define HCP_CODEC_HEVC                          0
+#define HCP_CODEC_VP9                           1
```
Comment (QuPengfei) [inline src/i965_defines.h]: the   HCP_CODEC_VP9 is defined twice

File: src/gen10_vdenc_vp9.c [hunk 4]
```diff
@@ -0,0 +1,4629 @@
...
+                                   struct gen10_vdenc_vp9_dys_curbe_parameters *curbe_param)
+{
+    struct gen10_vdenc_vp9_dys_curbe_data *curbe_cmd;
+
+    curbe_cmd = i965_gpe_context_map_curbe(gpe_context);
+
+    if (!curbe_cmd)
+        return;
+
+    memset(cur
```
Comment (QuPengfei) [inline src/gen10_vdenc_vp9.c]: it is  good remove the commented line if it is not needed

**Comment by xhaihao** [comment]
Update, thx @QuPengfei

**Pull request merged**